src/components/PostList/PostList.tsx

```diff
@@ -1,5 +1,44 @@
 import React from 'react';
 
-export const PostList: React.FC = () => (
-  <>Put the list here</>
+import { PostInfo } from '../PostInfo';
+import { Post } from '../../types/Post';
+import { User } from '../../types/User';
+import { Comment } from '../../types/Comment';
+
+type Props = {
+  posts: Post[],
+  comments: Comment[],
+  users: User[],
+};
+
+export const PostList: React.FC<Props> = ({
+  posts,
+  users,
+  comments,
+}) => (
+  <div className="PostList">
+    {posts.map(postItem => {
+      const user = users.find(
+        person => person.id === postItem.userId,
+      );
+
+      const postComments = comments.filter(item => postItem.id === item.postId);
+
+      if (!user) {
+        return null;
+      }
+
+      const postWithComments = {
+        ...postItem,
+        user,
+        comments: postComments,
+      };
+
```

Comment:
It would be better to get all the necessary data outside `map` 
You can create a separate function to prepare an array with posts and use this array for mapping